### PR TITLE
Add HTTP success tests based on ASIO sockets

### PR DESCRIPTION
**Pull request by FolMing**

### CODE REVIEW

File: test/http.cc [hunk 1]
```diff
@@ -29,6 +30,132 @@ const char* schemes[] = {"http://", "https://"};
 
 INSTANTIATE_TEST_SUITE_P(Schemes, HTTPTest, testing::ValuesIn(schemes));
 
+
+TEST_P(HTTPTest, GetSuccessASIO) {
+  std::string scheme = GetParam();
+
+  // Currently we don't provide tests for https.
```
Comment (rjhuijsman) [inline test/http.cc:37]: Can you expand the comment to explain why, and when we should be able to add https tests?

File: test/http.cc [hunk 2]
```diff
@@ -29,6 +30,132 @@ const char* schemes[] = {"http://", "https://"};
...
+    asio::read_until(sock, request_buffer, "\r\n\r\n", ec);
+    EXPECT_FALSE(ec);
+
+    // Content
+    asio::read_until(sock, request_buffer, "\r\n\r\n", ec);
+    EXPECT_FALSE(ec);
+
+    asio::write(sock, asio::buffer(response, response.size()), ec);
+    EXPECT_FALSE(ec);
+  });
```
Comment (rjhuijsman) [inline test/http.cc:137]: This all looks veeerrrry similar to what we have in the GET test. Could we factor this out into a function that the two functions share?

**Review by rjhuijsman — CHANGES_REQUESTED** [review]
This is very helpful! Just two readability requests.

File: test/http.cc (hunk 2 above)
Comment (benh) [reply test/http.cc:137]: Agree, let's put this into the `HTTPTest` as that's where we'll do it in the future as well!

**Review by benh — CHANGES_REQUESTED** [review]

**Comment by FolMing** [comment]
Will be added in #178

**Pull request closed without merging**